site/content/docs/main/troubleshooting/envoy-container-draining.md

```diff
@@ -0,0 +1,28 @@
+# Envoy container stuck in unready/draining state
+
+It's possible for the Envoy containers to become stuck in an unready/draining state.
+This is an unintended side effect of the shutdown-manager sidecar container being restarted by the kubelet.
+For more details on exactly how this happens, see [this issue][1].
+
+If you observe Envoy containers in this state, you should `kubectl delete` them to allow new Pods to be created to replace them.
+
+To make this issue less likely to occur, you should:
+- ensure you have [resource requests][2] on all your containers
+- ensure you do **not** have a liveness probe on the shutdown-manager sidecar container in the envoy daemonset (this was removed from the example YAML in Contour 1.24.0).
+
+If the above are not sufficient for preventing the issue, you may also add a liveness probe to the envoy container itself, like the following:
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /ready
+    port: 8002
+  initialDelaySeconds: 15
+  periodSeconds: 5
+  failureThreshold: 6
+```
+
+This will help the envoy container recover if it does become stuck in this state.
```

Comment:
maybe note here that recover = restart/be recreated

i forget off the top of my head how graceful draining will/will not work

Reply:
> i forget off the top of my head how graceful draining will/will not work in this case

Yeah, it depends on what the exact sequence of operations was that preceded the container getting stuck in this state. If it was the shutdown-manager sidecar being restarted immediately beforehand, then _that_ would have triggered a graceful drain, followed by the liveness probe then restarting the envoy container, which is more or less OK, just stretched out over a slightly longer time frame than if the whole Pod had been terminated.

Reply:
I added a vague comment related to this